### PR TITLE
feat: Use a venv with halcyon

### PR DESCRIPTION
**Pull request by retr0h**
The instructions require a minimum version of Ansible.  Enforce these
dependencies through a venv.  Particularly useful when switching between
halcyon and other projects.  Also, useful for the future as Halcyon expands,
and may require additional pypi dependencies.

---
This change is [Reviewable](https://reviewable.io/reviews/att-comdev/halcyon-kubernetes/46)

### CODE REVIEW

**Comment by v1k0d3n** [comment]
@retr0h very nice addition, and very welcome! thank you for checking out, using and  submitting to the project. LGTM

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+	if [ ! -d venv ]; then \
+	    virtualenv --no-setuptools venv; \
+	fi
+	. venv/bin/activate; pip install -r requirements.txt
+
+clean:
+	$(RM) -rf venv
+	find . -name "*.pyc" -exec $(RM) -rf {} \;
+
+.PHONY:clean venv
```
Comment (v1k0d3n) [inline Makefile:13]: i was thinking about doing this at one point, and then got distracted on other projects. very clean change.

**Review by v1k0d3n — APPROVED** [review]

**Comment by aric49** [comment]
Testing your changes locally, i am getting the following errors when I run make.. Looks like the virtualenv might require setuptools to run?

```
aric@host:~/tmp/halcyon-kubernetes$ make
if [ ! -d venv ]; then \
    virtualenv --no-setuptools venv; \
fi
New python executable in /home/aric/tmp/halcyon-kubernetes/venv/bin/python
Installing pip, wheel...done.
. venv/bin/activate; pip install -r requirements.txt
Collecting ansible==2.2.1 (from -r requirements.txt (line 1))
  Using cached ansible-2.2.1.0.tar.gz
Could not import setuptools which is required to install from a source distribution.
Please install setuptools.
Makefile:4: recipe for target 'venv/bin/activate' failed
make: *** [venv/bin/activate] Error 1
```

I have the following versions of setuptools and virtualenv on my laptop in my parent Python installation:

```
virtualenv (15.1.0)
setuptools (26.1.1)
```

**Comment by v1k0d3n** [comment]
@aric49 where are we on the jenkins pipeline for Halcyon? i think the team is ready to add this back, correct?

**Comment by retr0h** [comment]
> Testing your changes locally, i am getting the following errors when I run make.. Looks like the virtualenv might require setuptools to run?

This looks more like a problematic python/pip/venv environment than a problem with this patchset.

**Comment by aric49** [comment]
~~@retr0h -- I wouldn't be surprised if this was an issue on my end... I've had a lot of issues with Python/Pip dependencies on my workstation.   @v1k0d3n Can you validate this works for you, and if so, then we can merge.~~

I retract my last comment. I got the venv configuration running by uninstalling and re-installing the virtualenv pip package:

`pip uninstall virtualenv; pip install virtualenv`

LGTM.   Good to merge.

File: README.md [hunk 2]
```diff
@@ -26,8 +26,11 @@ This project was originally implemented as a Vagrant deployment. If you are look
     - kube-workers (as many as you'd like, excluding the master)
     - kube-control (same as your master)
 2. Make any needed/custom modifications to the included `kube-deploy/group_vars/all.yml` configuration file.
-3. From the `kube-deploy` folder, use Ansible to deploy your cluster: `ansible-playbook -i ../inventory.ini kube-deploy.yml`
-
+3. From the root of the project:
```
Comment (v1k0d3n) [inline README.md]: this makes the assumptions that some pre-requisites are installed prior to make and venv activation. can you outline in the docs what packages need to be installed for users of the project? (i.e. `pip install virtualenv`)

**Review by v1k0d3n — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (retr0h) [reply README.md]: Updated.

**Comment by v1k0d3n** [comment]
thanks for the update. LGTM.

**Pull request merged**